blog/docs/articles/llm-agents-overview.md

```diff
@@ -0,0 +1,170 @@
+# LLM Agents with Knative
+
+**Author: [Calum Murray](https://www.linkedin.com/in/calum-ra-murray/), Software Engineering Intern @ Red Hat**
+
+LLMs are a transformative technology, enabling new forms of interactions between users and software systems.
+But how can we make sure that the LLM gives correct answers to users? Can we use LLMs as more than just a
+question answering/summarization tool and have them take actions on our behalf?
+
+LLM Agents and Tool Calling are some emerging patterns in the space that allow for LLMs to interact with
+other tools and models, solving much of the problem of correctness and giving LLMs the power to take actions
+for users. With all of these benefits, we have been looking in the Knative community for ways that we can
+enable you to build LLM agent systems more easily and in a method that is more in line with how you build
+the rest of your system: declarative and cloud native. In this blog post we are going to first go over some
+background on what exactly an LLM Agent is and how LLMs can call tools, and then we will discuss where we
+picture Knative fitting into this paradigm.
+
+## What is an Agent?
+
+In AI, an agent is defined as a system that is able to take in information about its environment and use
+that information to make decisions and take actions to accomplish a goal. In the context of LLMs, an Agent
+is a system with an LLM at its core that is able to make decisions on what actions to take as it works to
+answer the prompt it received. The most common actions LLM agents can be built to take are: sending text
+or other media to the user, calling a tool to help answer the user, and calling another agent to help answer
+the user. Generally speaking, an LLM agent will also have a system prompt explaining what it’s role is and
+giving it some rules over when to call tools and/or reply to the user. For most Agents, the control flow
+can be shown as follows:
+
+<figure markdown="1">
+![image](./images/llm-agent-flow.webp)
```

Comment:
does not render in preview site

Reply:
ah, we were trying to center it, guess this doesn't work :(